### PR TITLE
Add support for split and merge workflow patterns

### PR DESCRIPTION
**Pull request by sliverc**
* To split workflow a list of tasks is returned in jexl expression
* To merge multiple tasks of a workflow link to same flow

### CODE REVIEW

File: caluma/workflow/tests/test_work_item.py [hunk 1]
```diff
@@ -90,3 +98,109 @@ def test_complete_work_item_with_next(
...
+            }
+          }
+        }
+    """
+
+    inp = {"input": {"id": work_item_1.pk}}
+    result = schema_executor(query, variables=inp)
+
+    assert not result.errors
+    assert case.work_items.filter(status=models.WorkItem.STATUS_READY).count() == 1
```
Comment (winged) [inline caluma/workflow/tests/test_work_item.py]: I'd like to see some of the asserts before the to-be-tested operation. This helps the reader understand exactly what's supposed to happen, without inferring the state from the setup code. For example:
```python
assert ready_item_count == 2
operation_that_completes_an_item()
assert ready_item_count == 1
```

File: caluma/workflow/jexl.py [hunk 2]
```diff
@@ -9,8 +10,19 @@ class FlowJexl(JEXL):
...
+        self.add_transform("tasks", lambda spec: spec)
 
     def extract_tasks(self, expr):
-        return self.analyze(
+        yield from self.analyze(
             expr, partial(ExtractTransformSubjectAnalyzer, transforms=["task"])
         )
+
+        # tasks transforms return a list of literals
+        yield from [
```
[the remark this answers is not in the document]
Comment (sliverc) [reply caluma/workflow/jexl.py]: Done

File: caluma/workflow/migrations/0006_auto_20181207_1512.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                    models.UUIDField(
+                        default=uuid.uuid4,
+                        editable=False,
+                        primary_key=True,
+                        serialize=False,
+                    ),
+                ),
+            ],
+        ),
+        migrations.AlterUniqueTogether(name="flow", unique_together=set([])),
```
[the remark this answers is not in the document]
Comment (sliverc) [reply caluma/workflow/migrations/0006_auto_20181207_1512.py]: Yes this removes the unique together index.

File: caluma/workflow/tests/test_jexl.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import pytest
+
+from ..jexl import FlowJexl
+
+
+@pytest.mark.parametrize(
+    "expression,expected_tasks",
+    [("63 > 62 ? 'test'|task : ['test2', 'test3']|tasks", {"test", "test2", "test3"})],
+)
```
[the remark this answers is not in the document]
Comment (sliverc) [reply caluma/workflow/tests/test_jexl.py]: I have added some more sensible test cases.

File: caluma/workflow/tests/test_work_item.py (hunk 1 above)
Comment (sliverc) [reply caluma/workflow/tests/test_work_item.py]: Adjusted test case and also added some comments to hopefully make the test more comprehensible.

**Comment by sliverc** [comment]
@winged I have added some comments in the test case. Concerning documentation this is an open issue on my side and will be my next task to follow up so I would rather do it in a another PR.

Documentation PR will then provide example graphql files, describing of main models and expanding GraphQL schema documentation.

**Pull request merged**